x/consortium/client/cli/tx.go

```diff
@@ -0,0 +1,211 @@
+
+			msg, err := govtypes.NewMsgSubmitProposal(content, deposit, from)
+			if err != nil {
+				return err
+			}
+
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	cmd.Flags().String(cli.FlagTitle, "", "title of proposal")
+	cmd.Flags().String(cli.FlagDescription, "", "description of proposal")
+	cmd.Flags().String(cli.FlagDeposit, "", "deposit of proposal")
+
+	return cmd
+}
+
+// NewCmdSubmitUpdateValidatorAuthsProposal implements the command to submit an update-validator-auths proposal
+func NewCmdSubmitUpdateValidatorAuthsProposal() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update-validator-auths",
+		Args:  cobra.NoArgs,
+		Short: "Submit an update validator auths proposal",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Submit an update validator auths proposal.
+
+Example:
+$ %s tx gov submit-proposal update-validator-auths [flags]
+`,
+				version.AppName,
+			),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+			from := clientCtx.GetFromAddress()
+
+			depositStr, err := cmd.Flags().GetString(cli.FlagDeposit)
+			if err != nil {
+				return err
+			}
+
+			deposit, err := sdk.ParseCoinsNormalized(depositStr)
+			if err != nil {
+				return err
+			}
+
+			title, err := cmd.Flags().GetString(cli.FlagTitle)
+			if err != nil {
+				return err
+			}
+
+			description, err := cmd.Flags().GetString(cli.FlagDescription)
+			if err != nil {
+				return err
+			}
+
+			parseCommaSeparated := func(concat string) []string {
+				if concat == "" {
+					return []string{}
+				}
+				return strings.Split(concat, ",")
+			}
+
+			addingValidatorsStr, err := cmd.Flags().GetString(FlagAllowedValidatorAdd)
+			if err != nil {
+				return err
+			}
+			addingValidators := parseCommaSeparated(addingValidatorsStr)
+
+			deletingValidatorsStr, err := cmd.Flags().GetString(FlagAllowedValidatorDelete)
+			if err != nil {
+				return err
+			}
+			deletingValidators := parseCommaSeparated(deletingValidatorsStr)
+
+			createAuths := func(addings, deletings []string) ([]*types.ValidatorAuth, error) {
+				var auths []*types.ValidatorAuth
+
+				// check duplications
+				usedAddrs := map[string]bool{}
+				for _, addr := range append(addings, deletings...) {
+					if usedAddrs[addr] {
+						return auths, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "multiple auths for same validator: %s", addr)
+					}
+					usedAddrs[addr] = true
+				}
+
+				for _, addr := range addings {
+					auth := &types.ValidatorAuth{
+						OperatorAddress: addr,
+						CreationAllowed: true,
+					}
+					auths = append(auths, auth)
+				}
+				for _, addr := range deletings {
+					auth := &types.ValidatorAuth{
+						OperatorAddress: addr,
+						CreationAllowed: false,
+					}
+					auths = append(auths, auth)
+				}
```

Comment:
Why don't ValidatorAuth clear the address and just set CreationAllowed to false?

Reply:
May I ask you more detailed explanation on it?

Reply:
The CreationAllowed value seems to indicate if that OperatorAddress can be created as a validator.
However, I think it should be whether the address is registered as a key in the store, not whether it is determined through a boolean variable.
In other words, the CreationAllowed variable is unnecessary, and I think that the criterion for judging whether the validator can be created is based on whether the address is registered as a key in the store.

Reply:
It's very good question. As you can see, I'm also thinking about replacing that structure into a boolean value or just place holder (the first comment in my self-review). If consortium module has few features, then I will change the states. However, if the requirements involve many authorization flags on validators (e.g. UnbondAllowed, DelegationAllowed, ...), then to simplify queries and setters, it will remain.
This PR is not final, there are one or two refactorings on my schedule.

Reply:
Thanks for the great answer. 👍